GVFS/GVFS.Common/GitHubUpgrader.cs

```diff
@@ -523,7 +523,27 @@ public bool TryLoad(out string error)
                 this.UpgradeRing = RingType.NoConfig;
 
                 string ringConfig = null;
-                if (this.LocalConfig.TryGetConfig(GVFSConstants.LocalGVFSConfig.UpgradeRing, out ringConfig, out error))
+                string upgradeFeedUrl = null;
+
+                // 1. Try to read ring from upgrade.feedurl first.

```

Comment:
Style suggestion:  I think the comment here (and the `2. If feedurl didn't ...` comment below) would be more useful as a single comment here that describes the rules for determining the ring.

As they are now they're not adding much detail beyond what's already self-documented by the code.

Reply:
consolidated all comments into a single block.